llvm/lib/Transforms/Vectorize/VPRecipeBuilder.h

```diff
@@ -113,18 +140,34 @@ class VPRecipeBuilder {
 
 public:
   VPRecipeBuilder(VPlan &Plan, Loop *OrigLoop, const TargetLibraryInfo *TLI,
+                  const TargetTransformInfo *TTI,
                   LoopVectorizationLegality *Legal,
                   LoopVectorizationCostModel &CM,
                   PredicatedScalarEvolution &PSE, VPBuilder &Builder)
-      : Plan(Plan), OrigLoop(OrigLoop), TLI(TLI), Legal(Legal), CM(CM),
-        PSE(PSE), Builder(Builder) {}
+      : Plan(Plan), OrigLoop(OrigLoop), TLI(TLI), TTI(TTI), Legal(Legal),
+        CM(CM), PSE(PSE), Builder(Builder) {}
+
+  std::optional<PartialReductionChain>
+  getScaledReductionForInstr(const Instruction *ExitInst) {
+    auto It = ScaledReductionExitInstrs.find(ExitInst);
+    return It == ScaledReductionExitInstrs.end()
+               ? std::nullopt
+               : std::make_optional(It->second);
+  }
+
+  void addScaledReductionExitInstrs(SmallVector<PartialReductionChain> Chains);
```

Comment:
```suggestion
  void addScaledReductionExitInstrs(ArrayRef<PartialReductionChain> Chains);
```

Avoid passing SmallVector by value, better to use ArrayRef<PartialReductionChain> or const SmallVector<> &

Reply:
Not necessary now that `collectScaledReductions` exists.